File: src/main/java/com/mhl/mycompanybackend/models/Roles.java
package com.mhl.mycompanybackend.models;

import javax.persistence.*;

@Entity
@Table(name = "roles", uniqueConstraints = {
        @UniqueConstraint(columnNames = "role_name")
})
public class Roles {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(name = "role_name")
    private String RoleName;
    @ManyToOne
    @JoinColumn(name = "permission_id")
    private Permissions permissions;

    public Roles() {
    }

    public Roles(String roleName, Permissions permissions) {
        RoleName = roleName;
        this.permissions = permissions;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getId() {
        return id;
    }

    public String getRole_name() {
        return RoleName;
    }

    public void setRole_name(String role_name) {
        this.RoleName = role_name;
    }

    public Permissions getPermissions() {
        return permissions;
    }

    public void setPermissions(Permissions permissions) {
        this.permissions = permissions;
    }
}

File: src/main/kotlin/com/mhl/mycompanybackend/model/Roles.kt
package com.mhl.mycompanybackend.model

import javax.persistence.*

@Entity
@Table(name = "roles", uniqueConstraints = [UniqueConstraint(columnNames = ["role_name"])])
class Roles {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null

    @Column(name = "role_name")
    var role_name: String? = null

    @JvmField
    @ManyToOne
    @JoinColumn(name = "permission_id")
    var permissions: Permissions? = null

    constructor()
    constructor(roleName: String?, permissions: Permissions?) {
        role_name = roleName
        this.permissions = permissions
    }
}
